### PR TITLE
[pravila00-make-vector] New port

### PR DESCRIPTION
**Pull request by Pravila00**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] <del> The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines. </del>
- [ ] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx] (https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/make-vector/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,16 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    GITHUB_HOST https://gist.github.com/
+    REPO tcbrindle/e55bea06d4e3ac1f603f9989d44dceb9
+    REF 5429b40957ebc3411f6e9ce1c0f11db3191432d4
+    SHA512 027371d143108b9a39e8d807156ac3ea0197ebe9157144ac8f50c0a1a48105fd6f08fb370fe16fa01c4222dbb1ea8e79292c6de74e5e186cfa3e6ebe8449f64a
+    HEAD_REF master
+)
+
+file(COPY ${SOURCE_PATH}/make_vector.hpp DESTINATION ${CURRENT_PACKAGES_DIR}/include/make-vector)
```
Comment (Adela0814) [inline ports/make-vector/portfile.cmake:10]:
```suggestion
file(COPY "${SOURCE_PATH}/make_vector.hpp" DESTINATION "${CURRENT_PACKAGES_DIR}/include/make-vector")
```

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by BillyONeal** [comment]
@tcbrindle putting this in a gist rather than a repo suggests that they don't expect it to be packaged and shipped like this. @tcbrindle what do you think?

**Comment by BillyONeal** [comment]
> * The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.

I do not believe this is true.

**Review by BillyONeal — CHANGES_REQUESTED** [review]
(see above)

**Comment by Pravila00** [comment]
@BillyONeal When I marked that task, I meant that I checked repology and I did not see any existing name for that component.

What do you mean saying that you do not believe it? Do you mean that the item doesn't exist in repology?

Sorry for the possible confusion!

**Comment by tcbrindle** [comment]
Also, because this was an example I apparently forgot to use include guards or #pragma once in the header, so that probably needs to be changed...

**Comment by BillyONeal** [comment]
At a minimum it needs to be `tcbrindle-make-vector` or something like that given that this name isn't clearly the right bit.

**Comment by Pravila00** [comment]
Thanks for the feedback on this @tcbrindle! 

If it is okay for you, then I will probably create a proper repository for this code, so we can maintain it and add the needed pragma/guards. 
I will give all the credits to you as the original author in that repository.

If you prefer creating it by yourself, please let me know, otherwise I will move on :)

**Comment by Pravila00** [comment]
Hey @tcbrindle @BillyONeal 

In the end: 
- I have created this [GitHub make-vector repository](https://github.com/Pravila00/make-vector) that contains the initial gist code from @tcbrindle , plus a #pragma once header.
- I have renamed the port to trcbrindle-make-vector as suggested by @BillyONeal .

Let me know if we are all on the same page for moving forward with this into Vcpkg, thanks a lot folks!

File: ports/tcbrindle-make-vector/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,11 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO Pravila00/make-vector
```
Comment (BillyONeal) [inline ports/tcbrindle-make-vector/portfile.cmake:3]: Now it should be `pravila00-make-vector` :)

File: ports/tcbrindle-make-vector/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,11 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO Pravila00/make-vector
+    REF 1518ac00adec9b13a645aee45ed1a36eb6ec1e98
+    SHA512 2be4af258ceeb71e990ecc5c1c2c269456f6621b7b6b8183ba4e29f5479c4c7a618bd0c737a8d66aa00052a710930ec2f2ee5bc925f7a627427ac90918b6d4fa
+    HEAD_REF main
+)
+
+file(COPY "${SOURCE_PATH}/include/make_vector.hpp" DESTINATION "${CURRENT_PACKAGES_DIR}/include/tcbrindle-make-vector")
```
Comment (BillyONeal) [inline ports/tcbrindle-make-vector/portfile.cmake:9]: The installation directory doesn't necessarily need to contain the ownership prefix if you want

**Comment by Pravila00** [comment]
Thanks for the review @BillyONeal !

I think everything should be fine right now, let me know if you want to suggest anything else, otherwise I guess we are good to move forward :)

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Pull request merged**